packages/backend/src/helpers/zod-utils.ts

```diff
@@ -9,3 +10,19 @@ export function ensureZodObjectKey<
 >(_schema: TZodObject, key: TKey): TKey {
   return key
 }
+
+/**
+ * Helper to extract a message containing details about the 1st Zod parsing
+ * error. If a path is available, it also puts the path in the error message.
+ */
+export function firstZodParseError(error: ZodError): string {
```

Comment:
nice helper function, just wanted to point out that you may want to check the error messaging with Stacey for example for errors like float vs integer 
<img width="829" alt="image" src="https://github.com/opengovsg/plumber/assets/65110268/93069578-7f3d-417e-b980-924629dab9bd">

Reply:
thanks! Will do but i think should be fine for now. zod errors are a different beast, we tackle in another project.